detection_rules/rule_validators.py

```diff
@@ -43,28 +43,34 @@ def validate(self, data: QueryRuleData, meta: RuleMeta) -> None:
             if package_integrations:
                 # validate the query against related integration fields
                 self.validate_integration(data, meta, package_integrations)
-            else:
-                for stack_version, mapping in meta.get_validation_stack_versions().items():
-                    beats_version = mapping['beats']
-                    ecs_version = mapping['ecs']
-                    err_trailer = f'stack: {stack_version}, beats: {beats_version}, ecs: {ecs_version}'
-
-                    beat_types, beat_schema, schema = self.get_beats_schema(data.index or [],
-                                                                            beats_version, ecs_version)
-
-                    try:
-                        kql.parse(self.query, schema=schema)
-                    except kql.KqlParseError as exc:
-                        message = exc.error_msg
-                        trailer = err_trailer
-                        if "Unknown field" in message and beat_types:
-                            trailer = f"\nTry adding event.module or event.dataset to specify beats module\n\n{trailer}"
-
-                        raise kql.KqlParseError(exc.error_msg, exc.line, exc.column, exc.source,
-                                                len(exc.caret.lstrip()), trailer=trailer) from None
-                    except Exception:
-                        print(err_trailer)
-                        raise
+
+            if beats.parse_beats_from_index(data.index or []):
+                # validate the query against fields within beats
+                self.validate_beats(data, meta)
+
+    def validate_beats(self, data: QueryRuleData, meta: RuleMeta) -> None:
```

Comment:
This validates _all_ sources (based on index). Calling it only when beats indexes are found will bypass validation for all non beats sources.

As a reminder, it validates against ECS by default, and then if a beats index and it is an unrecognized ECS field, it will compare those against the beats dataset. With the integration validation expansion, it should do full integration schema, regardless of ECS.

Reply:
so you should be able to move this up